lib/Textile/API/index.ts

```diff
@@ -235,60 +230,58 @@ class API {
     return result as string
   }
 
-  searchContacts = async (query: pb.IContactQuery, options: pb.IQueryOptions, handler: (contact: pb.IContact, local: boolean) => void): Promise<{}> => {
-    return new Promise(async (resolve, reject) => {
-      // internal contact search result handler
-      let stream: EmitterSubscription
-      let streamError: EmitterSubscription
-      // just a helper to dedup below
-      const cleanup = () => {
-        if (stream) {
-          stream.remove()
-        }
-        if (streamError) {
-          streamError.remove()
-        }
+  searchContacts = (query: pb.IContactQuery, options: pb.IQueryOptions, handler: (event?: ContactSearchResult | 'DONE', error?: any) => void) => {
```

Comment:
FYI... Integrating the new contact search into my Photos branch, I found the combo handler/promise API really hard to work with. I changed this to be a pure handler API that sends a 'DONE' event when the search is complete. Much cleaner to consume.

Reply:
👍